beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -193,17 +201,53 @@ func handlePostSSZ(
 	return true
 }
 
-func sszRequested(req *http.Request) bool {
+func sszRequested(req *http.Request) (bool, error) {
 	accept, ok := req.Header["Accept"]
 	if !ok {
-		return false
+		return false, nil
+	}
+	if len(accept) == 0 {
+		return false, nil
 	}
-	for _, v := range accept {
-		if v == "application/octet-stream" {
-			return true
+	types := strings.Split(accept[0], ",")
+	// match a number with optional decimals
+	regex, err := regexp.Compile("q=\\d(\\.\\d)?")
```

Comment:
This only needs to be compiled once for the life of the program - you can avoid recompiling on every call by declaring a variable above the function in package scope, using https://pkg.go.dev/regexp#MustCompile